src/mem/cache/prefetch/stride.cc

```diff
@@ -149,6 +149,11 @@ Stride::calculatePrefetch(const PrefetchInfo &pfi,
 
         // Hit in table
         int new_stride = pf_addr - entry->lastAddr;
+
+        // Do nothing on repeated memory access
```

Comment:
Why is this the correct approach? My initial thought is that deciding if 0-strides should not modify the confidence should at the very least be a configuration parameter.

If we conclude that 0 strides should not update the confidence the code in line 157 needs to be updated.

Reply:
I agree. I've modified the code and now the users can choose between the granularity of cache line or word address. Only if the granularity is set to cache lines, we will ignore zero strides

Reply:
Alright, I agree with/understand the change to line 162; however, I don´t understand why the zero-stride check only applies to the cache-line scenario. Could you clarify?

Reply:
The reason why I was treating the the zero stride case separately was because I was trying to guard against multiple accesses to the same cache line. This can only happen if the prefetcher is operating at the granularity of the cache lines. If it operating at the granularity of word addresses, then we don't care even if we observe zero strides